cpp/src/io/parquet/reader_impl.cu

```diff
@@ -1210,16 +1223,19 @@ void reader::impl::decode_page_data(hostdevice_vector<gpu::ColumnChunkDesc> &chu
     }
 
     size_t max_depth = _metadata->get_output_nesting_depth(chunks[c].src_col_schema);
+    chunk_offsets.push_back(chunk_off);
 
-    // allocate (gpu) an array of pointers to validity data of size : nesting depth
-    chunk_nested_valids.emplace_back(hostdevice_vector<uint32_t *>{max_depth});
-    hostdevice_vector<uint32_t *> &valids = chunk_nested_valids.back();
-    chunks[c].valid_map_base              = valids.device_ptr();
+    // get a slice of size `nesting depth` from `chunk_nested_valids` to store an array of pointers
```

Comment:
`output nesting depth` is a little more precise than `nesting depth` to distinguish between "depth of the output column hierarchy" and "depth of the parquet scheme" which are very rarely the same.

Reply:
I simply edited the previous comment to indicate the new behaviour. The [previous comment](https://github.com/rapidsai/cudf/blob/f117b68ca68faa0acd8f9fa485cad0aa63a0e543/cpp/src/io/parquet/reader_impl.cu#L1214) also used the terminology `nesting depth`. Since stub schema elements don't contribute separately to the depth and hence don't add to the number of mask/data pointers, the chance of misinterpreting this seems slim.